### ISSUE TITLE
Unable to install FOG 1.5.4 on Ubuntu Server 18.04.1 (lacking php-mysqlnd)

### ISSUE THREAD

**Issue by Nono-m0le** [FOGProject/fogproject#261]
Hi,

It's probably a duplicate of the (closed #258 ) but as it's a different issue, here is the issue : 

During installation of FOG 1.5.4 on Ubuntu Server 18.04.1; step to reproduce : 
**sudo ./installfog.sh**
Choice: [2] **2**
```
 *** Detected a potential need to reinstall apache and php files.
 *** This will remove the /etc/php* and /etc/apache2* directories
 ***  and remove/purge the apache and php files from this system.
 *** If you're okay with this please type Y, anything else will
 ***  continue the installation, but may mean you will need to
 ***  remove the files later and make proper changes as
 ***  necessary. (Y/N):
``` 
**Y**
`What type of installation would you like to do? [N/s (Normal/Storage)]` 
**N**

`What is the IP address to be used by this FOG Server? [192.168.6.200]`
**192.168.2.1**

```
  Would you like to change the default network interface from ?
  If you are not sure, select No. [y/N]
``` 
**y**
`  What network interface would you like to use?` 
**p5p1**
**../lib/common/functions.sh: line 149: /8: syntax error: operand expected (error token is "/8")**

`  Would you like to setup a router address for the DHCP server? [Y/n]` 
**n**

`  Would you like DHCP to handle DNS? [Y/n] `
**n**

`  Would you like to use the FOG server for DHCP service? [y/N] `
**y**

```
  This version of FOG has internationalization support, would
  you like to install the additional language packs? [y/N] 
```
**N**

```
 * Here are the settings FOG will use:
 * Base Linux: Debian
 * Detected Linux Distribution: Ubuntu
 * Server IP Address: 192.168.2.1
 * Server Subnet Mask: Error: fl is not recognized
 * Interface: p5p1
 * Installation Type: Normal Server
 * Internationalization: 0
 * Image Storage Location: /images
 * Using FOG DHCP: Yes
 * DHCP router Address:


 * Are you sure you wish to continue (Y/N)
``` 
**Y**

Ending up with : 
` * Installing now everything is updated........................Failed!`

Here is the 'interesting' part of **error_logs\fog_error_1.5.4.log**:
```
dpkg-query: no packages found matching php-mcrypt
php-mysqlnd-ms
php-mysqlnd-ms
Reading package lists...
Building dependency tree...
Reading state information...
Package php-mysqlnd is a virtual package provided by:
  php7.3-mysql 7.3.0~beta2-2+ubuntu18.04.1+deb.sury.org+1
  php7.2-mysql 7.2.9-1+ubuntu18.04.1+deb.sury.org+1
  php7.1-mysql 7.1.20-1+ubuntu18.04.1+deb.sury.org+1
  php5.6-mysql 5.6.37-1+ubuntu18.04.1+deb.sury.org+1
  php7.0-mysql 7.0.31-1+ubuntu18.04.1+deb.sury.org+1

E: Package 'php-mysqlnd' has no installation candidate
php-mysqlnd-ms
php-mysqlnd-ms
Reading package lists...
Building dependency tree...
Reading state information...
Package php-mysqlnd is a virtual package provided by:
  php7.3-mysql 7.3.0~beta2-2+ubuntu18.04.1+deb.sury.org+1
  php7.2-mysql 7.2.9-1+ubuntu18.04.1+deb.sury.org+1
  php7.1-mysql 7.1.20-1+ubuntu18.04.1+deb.sury.org+1
  php5.6-mysql 5.6.37-1+ubuntu18.04.1+deb.sury.org+1
  php7.0-mysql 7.0.31-1+ubuntu18.04.1+deb.sury.org+1

E: Package 'php-mysqlnd' has no installation candidate
dpkg-query: no packages found matching sysv-rc-conf
ii  tar            1.29b-2      amd64        GNU version of the tar archiving utility
ii  tftpd-hpa      5.2+20150808-1ubuntu3 amd64        HPA's tftp server
ii  tftp-hpa       5.2+20150808-1ubuntu3 amd64        HPA's tftp client
ii  unzip          6.0-21ubuntu1 amd64        De-archiver for .zip files
ii  vsftpd         3.0.3-9build1 amd64        lightweight, efficient FTP server written for security
ii  wget           1.19.4-1ubuntu2.1 amd64        retrieves files from the web
dpkg-query: no packages found matching xinetd
ii  zlib1g:amd64   1:1.2.11.dfsg-0ubuntu2 amd64        compression library - runtime
Reading package lists...
Building dependency tree...
Reading state information...
Package php-mysqlnd is a virtual package provided by:
  php7.3-mysql 7.3.0~beta2-2+ubuntu18.04.1+deb.sury.org+1
  php7.2-mysql 7.2.9-1+ubuntu18.04.1+deb.sury.org+1
  php7.1-mysql 7.1.20-1+ubuntu18.04.1+deb.sury.org+1
  php5.6-mysql 5.6.37-1+ubuntu18.04.1+deb.sury.org+1
  php7.0-mysql 7.0.31-1+ubuntu18.04.1+deb.sury.org+1

E: Package 'php-mysqlnd' has no installation candidate
Reading package lists...
Building dependency tree...
Reading state information...
Package php-mysqlnd is a virtual package provided by:
  php7.3-mysql 7.3.0~beta2-2+ubuntu18.04.1+deb.sury.org+1
  php7.2-mysql 7.2.9-1+ubuntu18.04.1+deb.sury.org+1
  php7.1-mysql 7.1.20-1+ubuntu18.04.1+deb.sury.org+1
  php5.6-mysql 5.6.37-1+ubuntu18.04.1+deb.sury.org+1
  php7.0-mysql 7.0.31-1+ubuntu18.04.1+deb.sury.org+1

E: Package 'php-mysqlnd' has no installation candidate
```
EDIT:
Btw, here is the `df -h` : 
```
Filesystem      Size  Used Avail Use% Mounted on
udev            1.9G     0  1.9G   0% /dev
tmpfs           382M  1.4M  381M   1% /run
/dev/sda2       458G  7.2G  428G   2% /
tmpfs           1.9G     0  1.9G   0% /dev/shm
tmpfs           5.0M     0  5.0M   0% /run/lock
tmpfs           1.9G     0  1.9G   0% /sys/fs/cgroup
/dev/loop0       87M   87M     0 100% /snap/core/4917
tmpfs           382M     0  382M   0% /run/user/1000
/dev/loop1       87M   87M     0 100% /snap/core/5145
```

**Comment by mastacontrola** [comment]
We have a running set of tests for installation on a few different operating systems. Those operating systems include Ubuntu 18.04, and we're not seeing any problems.

I'd be interested to know how  you configured Ubuntu during install?  I'm not saying there isn't a problem, just saying our tests aren't showing the same problem.

Is this an upgrade or fresh install? If it is an upgrade, would you mind opening the `/opt/fog/.fogsettings` file and remove the `packages=` line(s)?

**Comment by Nono-m0le** [comment]
Hi @mastacontrola ,

It's a fresh install, configuration is quite ... basic. Automatic formatting to use the whole disk (no separate partition / LVM or anything).
I'm using an appliance with 3 Ethernet ports (which may be unconventional), but well detected by Ubuntu.
Are we talking about the recent Ubuntu **Server** 18.04 **.1 LTS** from July 26 ?

I notice that the WORKING branch can be installed without issue (but the capture task is still not working on my case), but the MASTER failed. Could that be that your test are running on the WORKING and not the MASTER ?

**Comment by Nono-m0le** [comment]
Thanks @Quazz : but what should I say ?
I've downloaded the latest ISO from the ubuntu website, install it from crash on a server, and the fog installation doesn't work. Would that be possible to get the way you install your test machine in order to reproduce the environment ?

**Comment by Nono-m0le** [comment]
I think this matrix show Ubuntu DESKTOP flavor as currently being tested, but not the SERVER flavor.
Here, you can have both : http://releases.ubuntu.com/18.04/

**Comment by Sebastian-Roth** [comment]
@Nono-m0le The first thing that jumped at me is that you seem to change the IP address.

> `What is the IP address to be used by this FOG Server? [192.168.6.200]`
**192.168.2.1**

Can you please provide the output of the command `ip a s` on this system? I am wondering how many interfaces you have and why FOG is not able to figure out things properly? This is not actually causing the problem but I am fairly sure this would cause other problems later on!

For FOG to be able to get things right you need to have your network interface ready setup on linux. Just wanted to get this out because there could be some confusion on FOG might do the interface configuration for you. It does not!

Another thing I'd suggest: Use a proper root shell instead of running the installer as sudo command.
```
sudo -i
cd fogproject/bin
./installfog.sh
```

Now to the main problem that I see. I just did a manual install un Ubuntu 18.04 SERVER and had the same error that pas posted in #258:
```
The following packages have unmet dependencies:
 php7.1-mcrypt : Depends: libmcrypt4 but it is not installable
E: Unable to correct problems, you have held broken packages.
```
This is because Ubuntu 18.04 SERVER does not come with `universe` section enabled in the sources list. See the [package libmcrypt4](https://packages.ubuntu.com/bionic/libmcrypt4) is not part of the default `main` section. So after doing a clean Ubuntu install edit `/etc/apt/sources.list`, add the `universe` section, save the file and run FOG installer.

PS: I don't see any php-mysqlnd problems on my test install. See this installer is trying to make the impossible work and install FOG on several different Linux distributions and versions. So mysqlnd is from a time when this was the more appropriate mysql lib to use. We still have that in the scripts but from my experience this should not cause any harm on Ubuntu 18.04 (just tested).

**Comment by Nono-m0le** [comment]
Hi @Sebastian-Roth,

Thanks for the reply,
TL;DR : Adding the universe source was the main issue here. The other issue was the 2nd interfaces used wasn't "UP" (even though it was configured prior to the installfog script execution). I've also used "root" to execute the script instead of using SUDO which ultimately solve my issue.

So, as stated a bit above, I'm using a NUC Appliance with 3 eth. interfaces. At the installation of Ubuntu, they are all correctly detected as : 
enp0s25 (connected to my LAN Network, with DHCP) => 192.168.6.0/24
enp4s0 (my future LAB/FOG Network) => I'll create it on 192.168.2.1/24
enp5s0 => Unused

Indeed, BEFORE the execution of the script, the enp4s0 is "up" and setup as "static".
$ ip a s 
```
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host
       valid_lft forever preferred_lft forever
2: enp0s25: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP group default qlen 1000
    link/ether f4:4d:30:65:18:c9 brd ff:ff:ff:ff:ff:ff
    inet 192.168.6.128/24 brd 192.168.6.255 scope global dynamic enp0s25
       valid_lft 6987sec preferred_lft 6987sec
    inet6 fe80::f64d:30ff:fe65:18c9/64 scope link
       valid_lft forever preferred_lft forever
7: enp4s0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP group default qlen 1000
    link/ether f8:02:78:20:0b:28 brd ff:ff:ff:ff:ff:ff
    inet 192.168.2.1/24 brd 192.168.2.255 scope global enp4s0
       valid_lft forever preferred_lft forever
    inet6 fe80::fa02:78ff:fe20:b28/64 scope link
       valid_lft forever preferred_lft forever
8: enp5s0: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc mq state DOWN group default qlen 1000
    link/ether f8:02:78:20:0b:29 brd ff:ff:ff:ff:ff:ff
```

But this time with the universe, it works like a charm (detecting my interface + IP as well).
Many thanks !

**Comment by Sebastian-Roth** [comment]
@Nono-m0le I just figured what the difference is Ubuntu 18.04 server comes in two different flavors. One is the normal server ISO and the other one is the LIVE server. For 18.04 straight you could still download both versions - [see here](http://old-releases.ubuntu.com/releases/18.04.0/). But seems like since 18.04.1 is out there is only the LIVE server image available anymore. I just installed 18.04.0 plain server and it has the mentioned `universe` section in sources.list
